frame.go

```diff
@@ -462,51 +462,52 @@ type frame interface {
 	Header() frameHeader
 }
 
-func readHeader(r io.Reader, p []byte) (head frameHeader, err error) {
-	_, err = io.ReadFull(r, p[:1])
-	if err != nil {
-		return frameHeader{}, err
+func readHeader(connProto byte, r io.Reader, p []byte) (head frameHeader, err error) {
+	if connProto == 2 {
```

Comment:
Why even support v2?

Reply:
That's OK - but those people can  (and maybe should!) stay with older drivers as well, as this is not a tested code path, most likely.